### PR TITLE
Jmw/week util methods

### PR DESCRIPTION
**Pull request by jonworek**
I've added a few helper methods that allow me to get a start/end date for a given merch week.  I need this to query AA's actualized data api for the rof.

Our UI uses merch weeks/months/quarters, and AA's API likes date ranges.

### CODE REVIEW

File: lib/merch_calendar/date_calculator.rb [hunk 1]
```diff
@@ -49,6 +51,17 @@ def end_of_month(year, merch_month)
...
 
+    def start_of_week(year, month, merch_week)
+      weeks = MerchCalendar.weeks_for_month(year, month)
+
+      week = weeks.find { |week| week.week == merch_week }
+      week.start_of_week
+    end
+
+    def end_of_week(year, month, merch_week)
+      start_of_week(year, month, merch_week) + 6
```
Comment (davidamcclain) [inline lib/merch_calendar/date_calculator.rb]: What's the `6` about? Can we define/name magic numbers please?

File: lib/merch_calendar/date_calculator.rb [hunk 2]
```diff
@@ -49,6 +51,17 @@ def end_of_month(year, merch_month)
       end
     end
 
+    def start_of_week(year, month, merch_week)
+      weeks = MerchCalendar.weeks_for_month(year, month)
+
+      week = weeks.find { |week| week.week == merch_week }
```
Comment (davidamcclain) [inline lib/merch_calendar/date_calculator.rb]: This is a bit smelly. Do you have any other ideas how to go about this?

File: lib/merch_calendar/date_calculator.rb (hunk 1 above)
Comment (jonworek) [inline lib/merch_calendar/date_calculator.rb]: Lol, you should see the rest of this file beyond just this small diff.  One might say I was keeping consistent with the style of the code around it 😉 

But, point taken.  I can do better.  Will address.

File: lib/merch_calendar/date_calculator.rb (hunk 2 above)
Comment (jonworek) [inline lib/merch_calendar/date_calculator.rb]: Digging through the code, `MerchCalendar::MerchWeek.find(2017, 4, 1)` will give the same result.  So it'll clean this up here, though that code is doing something similar.

**Comment by jonworek** [comment]
After digging through the code a little bit more and discovering some new methods that I didn't know existed, I don't think this PR holds as much value as when I started out.

But, I still think the convenience I've added is nice.  @davidamcclain are you willing to give a 👍 ?

**Pull request merged**